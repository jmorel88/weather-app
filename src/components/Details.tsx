import { formatTime } from "../utils/datetime";
interface Props {
  feelsLikeTemp: number;
  windSpeed: number;
  humidity: number;
  visibility: number;
  sunrise: number;
  sunset: number;
}

function DetailBox({ title, value }: { title: string; value: string }) {
  return (
    <li className=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg">
      <h3 className="text-gray-500 text-lg mb-4">{title}</h3>
      <p className="text-xl">{value}</p>
    </li>
  );
}

export default function Details({
  feelsLikeTemp,
  windSpeed,
  humidity,
  visibility,
  sunrise,
  sunset,
}: Props) {
  return (
    <section data-testid="details" className="text-white-900 text-center md:text-left">
      <h2 className="text-2xl font-semibold mb-6">Weather Details</h2>
      <ul className="p-4 md:p-8 bg-black-200 rounded-lg grid grid-rows-3 md:grid-rows-2 grid-cols-2 md:grid-cols-3 gap-y-4 gap-x-6">
        <DetailBox title="Feels Like" value={`${Math.round(feelsLikeTemp)}Â°`} />
        <DetailBox title="Wind Speed" value={`${Math.round(windSpeed)}mph`} />
        <DetailBox title="Humidity" value={`${Math.round(humidity)}%`} />
        <DetailBox title="Visibility" value={`${visibility / 1000} Km`} />
        <DetailBox title="Sunrise" value={formatTime(sunrise)} />
        <DetailBox title="Sunset" value={formatTime(sunset)} />
      </ul>
    </section>
  );
}
