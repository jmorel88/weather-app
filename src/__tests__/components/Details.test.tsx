import { describe, test } from "vitest";
import { render } from "@testing-library/react";
import Details from "../../components/Details";
import { mockWeatherResponse as weather } from "../mocks";

describe("Details", () => {
  test("renders correctly", ({ expect }) => {
    const { container } = render(
      <Details
        feelsLikeTemp={weather.main.feels_like}
        windSpeed={weather.wind.speed}
        humidity={weather.main.humidity}
        visibility={weather.visibility}
        sunrise={weather.sys.sunrise}
        sunset={weather.sys.sunset}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
