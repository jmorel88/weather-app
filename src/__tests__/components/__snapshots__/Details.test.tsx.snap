// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Details > renders correctly 1`] = `
<div>
  <section
    class="text-white-900 text-center md:text-left"
    data-testid="details"
  >
    <h2
      class="text-2xl font-semibold mb-6"
    >
      Weather Details
    </h2>
    <ul
      class="p-4 md:p-8 bg-black-200 rounded-lg grid grid-rows-3 md:grid-rows-2 grid-cols-2 md:grid-cols-3 gap-y-4 gap-x-6"
    >
      <li
        class=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg"
      >
        <h3
          class="text-gray-500 text-lg mb-4"
        >
          Feels Like
        </h3>
        <p
          class="text-xl"
        >
          100Â°
        </p>
      </li>
      <li
        class=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg"
      >
        <h3
          class="text-gray-500 text-lg mb-4"
        >
          Wind Speed
        </h3>
        <p
          class="text-xl"
        >
          10mph
        </p>
      </li>
      <li
        class=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg"
      >
        <h3
          class="text-gray-500 text-lg mb-4"
        >
          Humidity
        </h3>
        <p
          class="text-xl"
        >
          100%
        </p>
      </li>
      <li
        class=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg"
      >
        <h3
          class="text-gray-500 text-lg mb-4"
        >
          Visibility
        </h3>
        <p
          class="text-xl"
        >
          0.1 Km
        </p>
      </li>
      <li
        class=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg"
      >
        <h3
          class="text-gray-500 text-lg mb-4"
        >
          Sunrise
        </h3>
        <p
          class="text-xl"
        >
          5:53 AM
        </p>
      </li>
      <li
        class=" text-center bg-black-100/50 p-4 md:p-6 rounded-lg"
      >
        <h3
          class="text-gray-500 text-lg mb-4"
        >
          Sunset
        </h3>
        <p
          class="text-xl"
        >
          8:10 PM
        </p>
      </li>
    </ul>
  </section>
</div>
`;
